main/getting-started/intro-zoe.md

```diff
@@ -1,358 +1,332 @@
-they have ten houses up for sale, they have ten different contract instances.
-
-In addition to an `installation`, when a contract is instantiated, Zoe needs to 
-know what kind of ERTP assets the contract will work with. In Atomic Swap’s test 
-code, the full instantiation call is something like:
+::: warning Note
+
+E() is part of the Agoric platform and is used to [call methods on
+remote
+objects and receive a promise for the result](/distributed-programming.md#communicating-with-remote-objects-using-e). From our perspective, Zoe is a
+remote object, so we must use E().
+:::
+
+The invitation contained information about the `installation`, the
+installed code object that was used to create the instance. We can
+quickly compare this installation to ones that we know using object
+identity. No need to compare line-by-line to ensure the code is the same.
+
 ```js
-const issuerKeywordRecord = harden({
-  Asset: quatloosKit.issuer,
-  Price: moolaKit.issuer,
-});
-const atomicSwapInstance = zoe.startInstance(installation, issuerKeywordRecord);
+const isCorrectCode = installation === atomicSwapInstallation;
+```
+However, if we don't recognize the installation, we can inspect the
+code directly by calling:
+
+```js
+const bundledCode = await E(installation).getBundle();
+```
+
+Contracts can add their own information to the invitation. In this
+case, the Atomic Swap contract adds information about what is being
+traded: the `asset`, the amount which Alice has escrowed, and the
+`price`, what we must pay to get the asset. Let's imagine that `asset`
+was an amount of 3 moola, and `price` was an amount of 7 simoleans.
+(Moola and simoleans are made-up currencies for the purpose of this
+example.) Amounts are descriptions of digital assets, but have no
+value themselves. [For more on amounts, please see the ERTP
+guide.](/ertp/guide/amounts.html#amounts-values-and-brands)
+
+### Making an offer
+
+Now that we've checked out the invitation, we can make an offer.
+
+There are three required parts to an offer:
+* a Zoe invitation
+* a proposal
+* ERTP payments of digital assets, to be escrowed
+
+The `proposal` is a statement of what you want from the offer, and
+what you are giving in return. Zoe will use the proposal as an
+invariant to ensure that we don't lose our assets in the trade. This
+invariant is known as offer safety.
+
+We can use the `asset` and `price` amounts from the invitation to make
+our proposal: 
+
+```js
+const proposal = {
+  want: { Asset: asset }, // asset: 3 moola
+  give: { Price: price }, // price: 7 simoleans
+};
+```
+
+Proposals must use Keywords, which are keys that are ASCII and
+capitalized. In this case, the keywords to use [were determined by the
+contract
+code](https://github.com/Agoric/agoric-sdk/blob/23c3f9df56940230e21a16b4861f40197192fdea/packages/zoe/src/contracts/atomicSwap.js#L29):
+Asset and Price.
+
+We said that we are giving 7 simoleans, so we must send 7 simoleans in
+the form of a ERTP payment. ([ERTP
```

Comment:
```suggestion
We said that we are giving 7 simoleans, so zoe will ensure that we include 7 simoleans in the 
the form of a ERTP payment. ([ERTP
```

Reply:
Hmm, I think this sounds like Zoe is helping us make the payment, rather than something that we must do and which Zoe will enforce. I didn't mean to say anything about Zoe here, but more that we need to make a payment.